### ISSUE TITLE
[Feature] Valid SQL PLugin

### ISSUE THREAD

**Issue by siddharthsambharia-portkey** [Portkey-AI/gateway#516]
### What Would You Like to See with the Gateway?

Plugin Name: Valid SQL
Plugin Description: Validates whether the given SQL code is syntactically correct

Here is the [Contributing Guide for Plugin](https://github.com/Portkey-AI/gateway/blob/main/plugins/Contributing.md)

**Comment by keshavkrishna** [comment]
@VisargD  can i work on this

**Comment by VisargD** [comment]
Sure. Assigning the issue to you. Thanks!

**Comment by 5war00p** [comment]
@vrushankportkey Looks like no activity here as well, can I take this up?

Question: What is the scope of this, do we consider about DB specific syntaxes here? or just a regular SQL only. And is it okay to use js sql parser kind of package for this.

I'm thinking to use [node-sql-parser](https://www.npmjs.com/package/node-sql-parser), as it comes with support of 13 DBs. But looks like bundle size is bit heavier: 
<img width="402" alt="image" src="https://github.com/user-attachments/assets/334f7c69-6aaf-476e-aec9-6b1d51b02dd2">.

**Comment by vrushankportkey** [comment]
Hi @5war00p please take it up! Yes we would like the bundle size to be as small as possible. @VisargD thoughts?

**Comment by 5war00p** [comment]
Yeah makes sense, I'm not sure whether it is tree-shakable or not. We should validate that as well, if we are choosing some package.

**Comment by narengogi** [comment]
closing as stale
